src/overlays/actors/ovl_En_Sth/z_en_sth.c

```diff
@@ -315,313 +346,322 @@ void func_80B67540(EnSth* this, PlayState* play) {
 
-                        switch (sp2C) {
-                            case 0:
-                                if (CHECK_WEEKEVENTREG(WEEKEVENTREG_13_40)) {
-                                    this->actor.home.rot.z = 6;
+                        switch (day) {
+                            case 0: // first day
+                                if (CHECK_WEEKEVENTREG(WEEKEVENTREG_RECEIVED_OCEANSIDE_WALLET_UPGRADE)) {
+                                    STH_GI_ID(&this->actor) = GI_RUPEE_SILVER;
                                 } else {
-                                    SET_WEEKEVENTREG(WEEKEVENTREG_13_40);
+                                    SET_WEEKEVENTREG(WEEKEVENTREG_RECEIVED_OCEANSIDE_WALLET_UPGRADE);
                                     switch (CUR_UPG_VALUE(UPG_WALLET)) {
                                         case 0:
-                                            this->actor.home.rot.z = 8;
+                                            STH_GI_ID(&this->actor) = GI_WALLET_ADULT;
                                             break;
 
                                         case 1:
-                                            this->actor.home.rot.z = 9;
+                                            STH_GI_ID(&this->actor) = GI_WALLET_GIANT;
                                             break;
                                     }
                                 }
                                 break;
 
-                            case 1:
-                                this->actor.home.rot.z = 5;
+                            case 1: // second day
+                                STH_GI_ID(&this->actor) = GI_RUPEE_PURPLE;
                                 break;
 
-                            default:
-                                this->actor.home.rot.z = 4;
+                            default: // final day
+                                STH_GI_ID(&this->actor) = GI_RUPEE_RED;
                                 break;
                         }
                         func_801477B4(play);
-                        this->actionFunc = func_80B67458;
-                        func_80B67458(this, play);
+                        this->actionFunc = EnSth_GiveOceanspiderhouseReward;
+                        EnSth_GiveOceanspiderhouseReward(this, play);
                         break;
 
-                    case 0x113C:
-                        func_80151938(play, 0x113B);
+                    case 0x113C:                     // (Second day) I am giving you my life savings
+                        func_80151938(play, 0x113B); // If only I had gotten here yesterday...
                         break;
 
-                    case 0x1141:
-                        func_80151938(play, 0x1140);
-                        func_80B670A4(this, 3);
+                    case 0x1141:                     // (Final day) This is all I have
+                        func_80151938(play, 0x1140); // If only I had gotten here two days ago...
```

Comment:
Are these exact text quotes?

Reply:
No in the past I was asked no never include exact quotes with the reasoning being they were assets, all of the text comments I add are simplified or diversified to avoid exact matches